### DISCUSSION TITLE
.NET MAUI Blazor hosting model support

### DISCUSSION THREAD

**Discussion by qunshen** [oqtane/oqtane.framework#2127] [General]
the availability of [.NET Multi-platform App UI](https://dotnet.microsoft.com/en-us/apps/maui) (.NET MAUI) Release Candidate has just been announced. Does Oqtane has plan to support NET MAUI?

**Comment by sbwalker** [comment]
Yes, Oqtane plans to support MAUI. MAUI is not really a new hosting model for Blazor - it is a way to expose your Blazor application in a WebView.

**Reply by qunshen** [reply]
great! is there a road map for this? I am looking forward...

**Reply by sbwalker** [reply]
Oqtane 3.1 was a major milestone release so I expect a number of minor releases in the coming months focusing on overall framework integrity, performance, and minor enhancements. Also keep in mind that .NET 6 is an LTS release - which provides the most security and support for users of the platform - whereas .NET 7 will be a GA release... so it is not clear whether it makes sense for Oqtane to move to .NET 7 this fall or wait for .NET 8. That being said, it may be possible to create a separate MAUI project on .NET 7 which provides integration with a .NET 6 version of Oqtane. Overall more research is required in this area... which is not surprising because MAUI itself does not yet have a firm GA release date.

**Reply by qunshen** [reply]
With MAUI BlazorWebView, Oqtane cab be real native app and  can be uploaded to app store for download! and this overcome the disadvanetage of long download time of webassembly! The long download time provent webassembly app from widespread use in real world apps. Oqtane  support for MAUI is very important for Oqtance to be used  large "scale up" apps .So support for MAUI sooner the better!

**Reply by wbonekamp** [reply]
I totally agree. We recently decided to fully move to Maui and try to migrate out Xamarin (forms) apps to Maui as soon as possible. Oqtane support for Maui would be great!

**Reply by sbwalker** [reply]
We definitely plan to support MAUI in Oqtane... it is in fact one of the key principles called out in the Oqtane Philosophy: https://www.oqtane.org/blog/!/20/oqtane-philosophy

"One of the benefits of the Blazor component model is that it is adaptable to a variety of different run-time scenarios. Currently it supports a server-side model via SignalR as well as a client-side model via WebAssembly. However, there have already been demonstrations of future options including desktop apps via Electron or a native shell, progressive web applications (PWA), and native mobile apps via Xamarin. The goal for Oqtane is to support all of these run-time scenarios. In order to accomplish this, Oqtane is built with a clearly defined client/server architecture. This architecture provides the flexibility to adapt to a variety of run-time configurations."

**Reply by Loclhh** [reply]
Sooo great information that Oqtane will support MAUI!!! Eager looking to that future! :)

**Comment by lefig** [comment]
Hi Shaun,

Does Oqtane currently support MAUI?

Thanks

**Comment by sbwalker** [comment]
@lefig yes, support for Blazor Hybrid (.NET MAUI) was added last year - please see https://www.youtube.com/watch?v=xyus7zlmgdc
